engine/runtime/src/main/java/org/enso/interpreter/runtime/error/Warning.java

```diff
@@ -206,4 +204,29 @@ boolean hasType() {
   Type getType(@CachedLibrary("this") TypesLibrary thisLib) {
     return EnsoContext.get(thisLib).getBuiltins().warning();
   }
+
+  private static Warning[] uniqueWarnings(Warning[] warnings) {
+    int uniqueLen = 0;
+    Warning last = null;
+    Warning[] tmp = new Warning[warnings.length];
+    System.arraycopy(warnings, 0, tmp, 0, warnings.length);
+    Arrays.sort(tmp, Comparator.comparing(Warning::getCreationTime).reversed());
+    for (int i = 0; i < tmp.length; i++) {
+      if (last == null || tmp[i].getCreationTime() != last.getCreationTime()) {
```

Comment:
Have we measured how many different warnings can we create at the same time?

Reply:
(Writing for the future archaeologists 🙂):

We discussed on Discord that `creationTime` is initialized with result of `EnsoContext::clockTick` which currently is:
```java
private final AtomicLong clock = new AtomicLong();

public long clockTick() {    
	return clock.getAndIncrement();
}
```
so this is not a timestamp but an always increasing counter - thus any two warning instances should have a different value (well at least until an overflow, but 2^64 sounds like a lot of warnings).

So we should be OK.